index.bs

```diff
@@ -7096,9 +7105,38 @@ Note that when |nonce| is empty, then the (signed) authenticator extension outpu
 
 Verifying the <code>[=devicePubKey=]</code> extension output is performed by the [=[RP]=] whenever a [=device public key=] is returned within the extension output.
 
+The [=devicePubKey=] extension adds the following [=struct/item=] to [=credential records=]:
+
+<dl dfn-for="credential record" dfn-type="abstract-op">
+    :   <dfn>devicePubKeys</dfn>
+    ::  A (possibly [=set/empty=]) [=set=] of [=device-bound key records=] associated with this [=public key credential source=].
+
+        A <dfn dfn>device-bound key record</dfn> is an abstract representation of a registered [=device-bound key=].
+        It is a [=struct=] with the following [=struct/items=]:
+
+        <dl dfn-for="devicePubKey record" dfn-type="abstract-op">
+            :   <dfn>aaguid</dfn>
+            ::  The [=AAGUID=] of the [=device-bound key=]'s [=managing authenticator=].
+                This MAY be different from the [=AAGUID=] in the [$credential record/attestationObject$], if any, of the containing [=credential record=].
+
+            :   <dfn>dpk</dfn>
+            ::  The public key portion of the [=hardware-bound device key pair=].
+
+            :   <dfn>scope</dfn>
+            ::  The scope of the [=device-bound key=]. See [[#sctn-device-publickey-extension-definition]] for details.
+
+            :   <dfn>fmt</dfn>
+            ::  The [=attestation statement format=] of the [=device-bound key=]'s [=attestation statement=].
+
+            :   <dfn>attStmt</dfn>
+            ::  The [=device-bound key=]'s [=attestation statement=].
+        </dl>
+</dl>
+
+
 ##### Registration (`create()`) ##### {#sctn-device-publickey-extension-verification-create}
 
-If the `devicePubKey` extension was included on a {{CredentialsContainer/create()|navigator.credentials.create()}} call, then the below verification steps are performed in the context of <a href=#reg-ceremony-verify-extension-outputs>this step</a> of [[#sctn-registering-a-new-credential]] using these variables established therein: |credential|, |clientExtensionResults|, |authData|, and |hash|. [=[RP]=] policy may specify whether a response without a `devicePubKey` is acceptable.
+If the `devicePubKey` extension was included on a {{CredentialsContainer/create()|navigator.credentials.create()}} call, then the below verification steps are performed in the context of <a href=#reg-ceremony-verify-extension-outputs>step 18</a> of [[#sctn-registering-a-new-credential]] using these variables established therein: |credential|, |clientExtensionResults|, |authData|, and |hash|. [=[RP]=] policy may specify whether a response without a `devicePubKey` is acceptable.
```

Comment:
>If the devicePubKey extension was included on a [navigator.credentials.create()](https://w3c.github.io/webappsec-credential-management/#dom-credentialscontainer-create) call

The RP may request a DPK via the extension, but the authenticator may not provide one. Is this referring to the request or response?

Do we need a statement in the processing logic for when the authenticator doesn't provide it back, or would that just be generic for all extensions?

Reply:
(and now in hindsight, I realize these comments are likely more appropriate for the DPK PR itself, not yours. let me know if I should move them)

Reply:
I think that's covered by the last part:

>[=[RP]=] policy may specify whether a response without a `devicePubKey` is acceptable.

But evidently it could still use some clarification, then. :) How about this?

```suggestion
If the `devicePubKey` extension was requested in a {{CredentialsContainer/create()|navigator.credentials.create()}} call, then the below verification steps are performed in the context of <a href=#reg-ceremony-verify-extension-outputs>step 18</a> of [[#sctn-registering-a-new-credential]] using these variables established therein: |credential|, |clientExtensionResults|, |authData|, and |hash|. [=[RP]=] policy may specify whether a response without a `devicePubKey` is acceptable.
```

Or perhaps even:

```suggestion
If the [=[RP]=] requested the `devicePubKey` extension in a {{CredentialsContainer/create()|navigator.credentials.create()}} call, then the below verification steps are performed in the context of <a href=#reg-ceremony-verify-extension-outputs>step 18</a> of [[#sctn-registering-a-new-credential]] using these variables established therein: |credential|, |clientExtensionResults|, |authData|, and |hash|. [=[RP]=] policy may specify whether a response without a `devicePubKey` is acceptable.
```

Reply:
@timcappalli Any thoughts on the above suggestions?

Reply:
@emlun the latter (If the RP requested) I think is the most clear.

Reply:
Thanks @timcappalli!

I also edited the last phrase to

>[...] without a `devicePubKey` extension output is acceptable.

I'll assume that's an acceptable edit too.